crate/cli/src/tests/datasets.rs

```diff
@@ -0,0 +1,176 @@
+use cosmian_logger::log_init;
+use std::collections::HashMap;
+use test_findex_server::start_default_test_findex_server;
+use uuid::Uuid;
+
+use crate::{
+    actions::datasets::{AddEntries, DeleteEntries, GetEntries},
+    cli_error,
+    error::{result::CliResult, CliError},
+    tests::{utils::recover_cmd_logs, PROG_NAME},
+};
+
+pub(crate) fn dataset_add_entries_cmd(
+    cli_conf_path: &str,
+    action: &AddEntries,
+) -> CliResult<String> {
+    let mut cmd = Command::cargo_bin(PROG_NAME)?;
+    let mut args = vec![
+        "add".to_owned(),
+        "--index-id".to_owned(),
+        action.index_id.to_string(),
+    ];
+    for (entry_id, data) in &action.entries {
+        args.push("-D".to_owned());
+        args.push(format!("{entry_id}={data}"));
+    }
+    cmd.env(FINDEX_CLI_CONF_ENV, cli_conf_path);
+
+    cmd.arg("datasets").args(args);
+    let output = recover_cmd_logs(&mut cmd);
+    if output.status.success() {
+        let findex_output = std::str::from_utf8(&output.stdout)?;
+        return Ok(findex_output.to_owned());
+    }
+    Err(CliError::Default(
+        std::str::from_utf8(&output.stderr)?.to_owned(),
+    ))
+}
+
+pub(crate) fn dataset_delete_entries_cmd(
+    cli_conf_path: &str,
+    delete_entries: &DeleteEntries,
+) -> CliResult<String> {
+    let mut cmd = Command::cargo_bin(PROG_NAME)?;
+    let mut args = vec![
+        "delete".to_owned(),
+        "--index-id".to_owned(),
+        delete_entries.index_id.to_string(),
+    ];
+    for uuid in delete_entries.uuids.clone() {
+        args.push("--uuids".to_owned());
+        args.push(uuid.to_string());
+    }
+    cmd.env(FINDEX_CLI_CONF_ENV, cli_conf_path);
+
+    cmd.arg("datasets").args(args);
+    let output = recover_cmd_logs(&mut cmd);
+    if output.status.success() {
+        let findex_output = std::str::from_utf8(&output.stdout)?;
+        return Ok(findex_output.to_owned());
+    }
+    Err(CliError::Default(
+        std::str::from_utf8(&output.stderr)?.to_owned(),
+    ))
+}
+
+pub(crate) fn datasets_get_entries_cmd(
+    cli_conf_path: &str,
+    get_entries: &GetEntries,
+) -> CliResult<EncryptedEntries> {
+    let mut cmd = Command::cargo_bin(PROG_NAME)?;
+    let mut args = vec![
+        "get".to_owned(),
+        "--index-id".to_owned(),
+        get_entries.index_id.to_string(),
+    ];
+    for uuid in get_entries.uuids.clone() {
+        args.push("--uuids".to_owned());
+        args.push(uuid.to_string());
+    }
+    cmd.env(FINDEX_CLI_CONF_ENV, cli_conf_path);
+
+    cmd.arg("datasets").args(args);
+    let output = recover_cmd_logs(&mut cmd);
+    if output.status.success() {
+        let findex_output = std::str::from_utf8(&output.stdout)?;
+        return parse_entries(findex_output);
+    }
+    Err(CliError::Default(
+        std::str::from_utf8(&output.stderr)?.to_owned(),
+    ))
+}
+
+#[allow(clippy::indexing_slicing)]
+fn parse_entries(s: &str) -> CliResult<EncryptedEntries> {
```

Comment:
Please confirm me that this is the right way to parse findex result. It works yes, but idk if my implementation is case specefic or would work on other datasets
Please also notice that there is a log duplication in findex raw output